server/pulp/server/controllers/importer.py

```diff
@@ -35,16 +37,28 @@ def build_resource_tag(repo_id, importer_type_id):
 
 def clean_config_dict(config):
     """
-    Remove keys from a dict that have a value None.
+    Ensure that the config is ready to be passed to the plugin. There are two things that must
+    occur:
+        1. Remove keys from the dict that have a value None.
+        2. If they exist, move basic auth credentials from the feed_url to the config.
 
     :param config: configuration for plugin
     :type  config: dict
 
     :return: config without the keys whose values were None
     :rtype:  dict:
     """
-
     if config is not None:
+        feed_url = config.get('feed')
+        if feed_url:
+            parsed = urlparse.urlparse(feed_url)
+            if parsed.username and parsed.password:
+                at = parsed.netloc.find('@')
+                new_url = parsed._replace(netloc=parsed.netloc[at + 1:])
```

Comment:
Updated, good catch! Please note that I had to make a new decision. If not present, `urlparse` sets the password to `None`, which will cause that key to be removed from the config in the next part of this function. If username is not present, `parsed.username` becomes the empty string.